stream_alert/apps/_apps/intercom.py

```diff
@@ -0,0 +1,103 @@
+import time
+import re
+
+import requests
+
+from . import AppIntegration, StreamAlertApp, get_logger
+
+
+LOGGER = get_logger(__name__)
+
+
+@StreamAlertApp
+class IntercomApp(AppIntegration):
+    """Intercom StreamAlert app"""
+    _INTERCOM_LOGS_URL = 'https://api.intercom.io/admins/activity_logs'
+
+    def __init__(self, event, config):
+        super(IntercomApp, self).__init__(event, config)
+        self._next_page = None
+
+    @classmethod
+    def _type(cls):
+        return 'admin_activity_logs'
+
+    @classmethod
+    def service(cls):
+        return 'intercom'
+
+    @classmethod
+    def _required_auth_info(cls):
+        return {
+            'token': {
+                'description': 'the OAuth token for this Intercom app',
+                'format': re.compile(r'.*')
```

Comment:
is there no static length or anything that could be used for the token to try to detect invalid input? a lot of services at least have a consistent length for their tokens, some even use a static prefix that can be helpful to validate.

Reply:
there doesn't seem to be a fixed length or prefix for the tokens. they are `base64` encoded though -- would a regex matching for a `base64` encoded string suffice?

Reply:
okay! all good to leave it as is :)

Reply:
actually found the source code and there is a prefix :) just updated the validation regex to look to for a base64 string with prefix!